Bottom: 4c4433d906639aed55de7876e8d857d10601b643
Top:    71cbde2dd6f2a685034d542258dd6cd8a40b470a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-20 23:30:30 +0100

Refresh of consolidate-pin-drawing-routin

---

diff --git a/src/crosshair.c b/src/crosshair.c
index f016094..f2ad726 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -353,13 +353,15 @@ XORDrawBuffer (BufferTypePtr Buffer)
   }
   END_LOOP;
 
-  /* and the vias, move offset by thickness/2 */
+  /* and the vias */
   if (PCB->ViaOn)
     VIA_LOOP (Buffer->Data);
   {
-    gui->draw_arc (Crosshair.GC,
-		   x + via->X, y + via->Y,
-		   via->Thickness / 2, via->Thickness / 2, 0, 360);
+    /* Make a copy of the via structure, moved to the correct position */
+    PinType moved_via = *via;
+    moved_via.X += x; moved_via.Y += y;
+
+    gui->thindraw_pcb_pv (Crosshair.GC, Crosshair.GC, &moved_via, false, false);
   }
   END_LOOP;
 }
